api/main.py

```diff
@@ -341,8 +362,18 @@ async def get_nodes_count(request: Request, kind: str = "node"):
         ) from error
 
 
+async def _verify_user_group_existence(user_groups: List[str]):
+    """Check if user group exists"""
+    for group_name in user_groups:
+        if not await db.find_one(UserGroup, name=group_name):
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail=f"User group does not exist with name: {group_name}")
+
+
 @app.post('/node', response_model=Node, response_model_by_alias=False)
-async def post_node(node: Node, token: str = Depends(get_user)):
+async def post_node(node: Node, user_groups: List[str] = [],
```

Comment:
BTW The linter warning here is about using `None` instead, as otherwise the same object reference keeps being used for default values